components/graphs/Graph.tsx

```diff
@@ -19,13 +19,29 @@ import {
 import useTime from '~/lib/hooks/useTime';
 import { ClearSkyDataPoint, ForecastDataPoint } from '~/lib/types';
 
-function getGraphStartDate(currentTime: number) {
+function getGraphStartDate(currentTime: number, totalHours: number) {
   const currentDate = new Date(currentTime);
   return new Date(
     currentDate.getFullYear(),
     currentDate.getMonth(),
     currentDate.getDate(),
-    currentDate.getHours() - 5
+    totalHours > 1
+      ? currentDate.getHours() - totalHours / 8
+      : currentDate.getHours(),
+    totalHours > 1 ? 0 : currentDate.getMinutes() - 15
```

Comment:
Could we perhaps get a comment above this function describing what this bit of math does?